lib/pavilion/test_run.py

```diff
@@ -781,15 +780,31 @@ def gather_results(self, run_result, regather=False, log_file=None):
         return results
 
     def save_results(self, results):
-        """Save the results to the results file.
+        """Save the results to the test specific resutls file and the general
+        pavilion results file.
 
-:param dict results: The results dictionary.
-"""
+        :param dict results: The results dictionary.
+        """
 
         with self.results_path.open('w') as results_file, \
                 PermissionsManager(self.results_path, self.group, self.umask):
             json.dump(results, results_file)
 
+        result_logger = logging.getLogger('common_results')
+        if self._pav_cfg.get('flatten_results') and results.get('per_file'):
+            # Flatten 'per_file' results into separate result records.
+            base = results.copy()
+            del base['per_file']
+
+            for per_file, values in results['per_file'].items():
+                per_result = base.copy()
+                per_result['file'] = per_file
+                per_result.update(values)
+
+                result_logger.info(output.json_dumps(per_result))
+        else:
```

Comment:
Is this strictly an either-or situation, or should this be done in both cases (minus the per_file content)?

Reply:
If there's no per_file content, there's nothing to flatten.